provider/search-provider/src/main/java/dev/ikm/tinkar/provider/search/TypeAheadSearch.java

```diff
@@ -0,0 +1,177 @@
+            reader.close();
+        }
+    }
+
+    public static List<String> suggest(String term) throws IOException {
+        List<Lookup.LookupResult> lookup = suggester.lookup(term, false, 100);
+        return lookup.stream().map(a -> a.key.toString()).collect(Collectors.toList());
+    }
+
+    public static void buildFuzzySuggester() throws IOException {
+        Stopwatch stopwatch = new Stopwatch();
+        reader = DirectoryReader.open(Indexer.indexWriter);
+        LuceneDictionary dict = new LuceneDictionary(reader, TEXT_FIELD_NAME);
+        fuzzySuggester = new FuzzySuggester(Indexer.indexDirectory, "suggest", Indexer.analyzer);
+        fuzzySuggester.build(dict);
+        LOG.info("TypeAheadSearch fuzzy index build duration: {}", stopwatch.durationString());
+    }
+
+    public static List<String> fuzzySuggest(String term) throws IOException {
+        List<Lookup.LookupResult> lookup = fuzzySuggester.lookup(term, false, 100);
+        return lookup.stream().map(a -> a.key.toString()).collect(Collectors.toList());
+    }
+
+    public static void buildSearchIndex(String term) throws IOException, ParseException, InvalidTokenOffsetsException {
+        DirectoryReader reader = DirectoryReader.open(Indexer.indexWriter);
+        LuceneDictionary dict = new LuceneDictionary(reader, TEXT_FIELD_NAME);
+        IndexSearcher indexSearcher = new IndexSearcher(reader);
+        QueryParser parser = new QueryParser("text", Indexer.analyzer());
+        Query query = parser.parse(term);
+        Formatter formatter = new SimpleHTMLFormatter();
+        QueryScorer scorer = new QueryScorer(query);
+        Highlighter highlighter = new Highlighter(formatter, scorer);
+        ScoreDoc[] hits = indexSearcher.search(query, 100).scoreDocs;
+        System.out.println(hits);
+        PrimitiveDataSearchResult[] results = new PrimitiveDataSearchResult[hits.length];
+
+        for (int i = 0; i < hits.length; i++) {
+            Document hitDoc = indexSearcher.doc(hits[i].doc);
+            StoredField nidField = (StoredField) hitDoc.getField(Indexer.NID);
+            StoredField patternNidField = (StoredField) hitDoc.getField(Indexer.PATTERN_NID);
+            StoredField rcNidField = (StoredField) hitDoc.getField(Indexer.RC_NID);
+            StoredField fieldIndexField = (StoredField) hitDoc.getField(Indexer.FIELD_INDEX);
+            StoredField textField = (StoredField) hitDoc.getField(Indexer.TEXT_FIELD_NAME);
+            String highlightedString = highlighter.getBestFragment(Indexer.analyzer(), Indexer.TEXT_FIELD_NAME, textField.stringValue());
+
+            results[i] = new PrimitiveDataSearchResult(nidField.numericValue().intValue(), rcNidField.numericValue().intValue(),
+                    patternNidField.numericValue().intValue(), fieldIndexField.numericValue().intValue(), hits[i].score, highlightedString);
+            System.out.println(results[i]);
+        }
+        System.out.println(results);
```

Comment:
System.out.printlns.

Reply:
these are in progress and will be addressed in the next typeahead task.